xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
If you looked at the code, you would see that there is no sync wanted. Assume one thread calls IsRunning() and the result is true. Task switch, thread stops, thread 1 continues, now IsRunning is not true anymore.
The way this code is written and used, the result of IsRunning may not be valid for the next operation. Given this fact, it does not matter if there is a task switch when while m_ThreadId.

You won't be able to construct a scenario where this change here would have any influence to current behaviour.

If you wanted to do something like this:
"Perform an operation only if thread is running and make running thread aware of this (i.e. prevent from terminating)"
you would need a lock over more than a single statement.